pkg/server/plugin/upstreamca/awssecret/awssecret_test.go

```diff
@@ -81,37 +81,35 @@ func (as *AWSSecretSuite) TestGetSecretFail() {
 }
 
 func (as *AWSSecretSuite) Test_SubmitValidCSR() {
-	const testDataDir = "_test_data/csr_valid"
-	validCsrFiles, err := ioutil.ReadDir(testDataDir)
+	validSpiffeID := "spiffe://localhost"
+	csr, pubKey, err := util.NewCSRTemplate(validSpiffeID)
 	as.Require().NoError(err)
 
-	for _, validCsrFile := range validCsrFiles {
-		csrPEM, err := ioutil.ReadFile(filepath.Join(testDataDir, validCsrFile.Name()))
-		as.Require().NoError(err)
-		block, rest := pem.Decode(csrPEM)
-		as.Require().Len(rest, 0)
+	resp, err := as.awsUpstreamCA.SubmitCSR(ctx, &upstreamca.SubmitCSRRequest{Csr: csr})
+	as.Require().NoError(err)
+	as.Require().NotNil(resp)
 
-		resp, err := as.awsUpstreamCA.SubmitCSR(ctx, &upstreamca.SubmitCSRRequest{Csr: block.Bytes})
-		as.Require().NoError(err)
-		as.Require().NotNil(resp)
-	}
+	cert, err := x509.ParseCertificate(resp.SignedCertificate.CertChain)
+	as.Require().NoError(err)
+
+	as.Require().True(cryptoutil.PublicKeyEqual(cert.PublicKey, pubKey))
```

Comment:
My suggestion looks like it was incorrect. `cryptoutil.PublicKeyEqual` returns `(bool, error)`. It looks like we're treating the error returned as the "message" for if the test fails. We should be checking the error (which you are already doing in upstreamca/disk.

Sorry!

Reply:
No problem :) 
Thank you for your suggestion. I will fix it!

Reply:
I fixed it with https://github.com/spiffe/spire/pull/988/commits/203fc0970968c4467bdccf2fbaee823c4d659c30